PM2020-9: convert a variant to simple product

nmarniesse opened the pull request:
Update product schema to explain the variant -> simple product conversion by using the `parent` field.

The swagger json was not up to date. Real diff:

```
<             "description": "Return product's quality scores in the response.",
---
>             "description": "Return product quality scores in the response. (Only available in the PIM Serenity version.)",
250c250
<                                     "description": "Code of the parent product model when the product is a variant (only available since the 2.0). This parent can be modified since the 2.3.",
---
>                                     "description": "Code of the parent product model when the product is a variant (only available since the 2.0). This parent can be modified since the 2.3. In Serenity and from the 5.0, the parent of an existing variant product can be set to null: the variant product will become a simple product (with no parent)",
405c405,407
<                                     "description": "Only available when query parameter \"with_quality_scores\" is set to \"true\""
---
>                                     "description": "Product quality scores for each channel/locale combination (only available in Serenity and when the \"with_quality_scores\" query parameter is set to \"true\")",
>                                     "x-immutable": true,
>                                     "x-read-only": true
549,552c551,555
<                                   "quality_scores": {
<                                     "print": {
<                                       "en_US": "A",
<                                       "fr_FR": "B"
---
>                                   "quality_scores": [
>                                     {
>                                       "scope": "ecommerce",
>                                       "locale": "en_US",
>                                       "data": "A"
554,556c557,565
<                                     "mobile": {
<                                       "en_US": "C",
<                                       "fr_FR": "D"
---
>                                     {
>                                       "scope": "ecommerce",
>                                       "locale": "fr_FR",
>                                       "data": "B"
>                                     },
>                                     {
>                                       "scope": "tablet",
>                                       "locale": "en_US",
>                                       "data": "D"
558,560c567,570
<                                     "ecommerce": {
<                                       "en_US": "A",
<                                       "fr_FR": "E"
---
>                                     {
>                                       "scope": "tablet",
>                                       "locale": "fr_FR",
>                                       "data": "E"
562c572
<                                   }
---
>                                   ]
738,749c748,767
<                     "quality_scores": {
<                       "print": {
<                         "en_US": "A",
<                         "fr_FR": "B"
<                       },
<                       "mobile": {
<                         "en_US": "C",
<                         "fr_FR": "D"
<                       },
<                       "ecommerce": {
<                         "en_US": "A",
<                         "fr_FR": "E"
---
>                     "quality_scores": [
>                       {
>                         "scope": "ecommerce",
>                         "locale": "en_US",
>                         "data": "A"
>                       },
>                       {
>                         "scope": "ecommerce",
>                         "locale": "fr_FR",
>                         "data": "B"
>                       },
>                       {
>                         "scope": "tablet",
>                         "locale": "en_US",
>                         "data": "D"
>                       },
>                       {
>                         "scope": "tablet",
>                         "locale": "fr_FR",
>                         "data": "E"
751c769
<                     }
---
>                     ]
846,857c864,883
<                     "quality_scores": {
<                       "print": {
<                         "en_US": "A",
<                         "fr_FR": "B"
<                       },
<                       "mobile": {
<                         "en_US": "C",
<                         "fr_FR": "D"
<                       },
<                       "ecommerce": {
<                         "en_US": "A",
<                         "fr_FR": "E"
---
>                     "quality_scores": [
>                       {
>                         "scope": "ecommerce",
>                         "locale": "en_US",
>                         "data": "A"
>                       },
>                       {
>                         "scope": "ecommerce",
>                         "locale": "fr_FR",
>                         "data": "B"
>                       },
>                       {
>                         "scope": "tablet",
>                         "locale": "en_US",
>                         "data": "C"
>                       },
>                       {
>                         "scope": "tablet",
>                         "locale": "fr_FR",
>                         "data": "D"
859c885
<                     }
---
>                     ]
1036c1062
<                   "description": "Code of the parent product model when the product is a variant (only available since the 2.0). This parent can be modified since the 2.3.",
---
>                   "description": "Code of the parent product model when the product is a variant (only available since the 2.0). This parent can be modified since the 2.3. In Serenity and from the 5.0, the parent of an existing variant product can be set to null: the variant product will become a simple product (with no parent)",
1191c1217,1219
<                   "description": "Only available when query parameter \"with_quality_scores\" is set to \"true\""
---
>                   "description": "Product quality scores for each channel/locale combination (only available in Serenity and when the \"with_quality_scores\" query parameter is set to \"true\")",
>                   "x-immutable": true,
>                   "x-read-only": true
1335,1346c1363,1382
<                 "quality_scores": {
<                   "print": {
<                     "en_US": "A",
<                     "fr_FR": "B"
<                   },
<                   "mobile": {
<                     "en_US": "C",
<                     "fr_FR": "D"
<                   },
<                   "ecommerce": {
<                     "en_US": "A",
<                     "fr_FR": "E"
---
>                 "quality_scores": [
>                   {
>                     "scope": "ecommerce",
>                     "locale": "en_US",
>                     "data": "A"
>                   },
>                   {
>                     "scope": "ecommerce",
>                     "locale": "fr_FR",
>                     "data": "B"
>                   },
>                   {
>                     "scope": "tablet",
>                     "locale": "en_US",
>                     "data": "D"
>                   },
>                   {
>                     "scope": "tablet",
>                     "locale": "fr_FR",
>                     "data": "E"
1348c1384
<                 }
---
>                 ]
1541c1577
<                   "description": "Code of the parent product model when the product is a variant (only available since the 2.0). This parent can be modified since the 2.3.",
---
>                   "description": "Code of the parent product model when the product is a variant (only available since the 2.0). This parent can be modified since the 2.3. In Serenity and from the 5.0, the parent of an existing variant product can be set to null: the variant product will become a simple product (with no parent)",
1696c1732,1734
<                   "description": "Only available when query parameter \"with_quality_scores\" is set to \"true\""
---
>                   "description": "Product quality scores for each channel/locale combination (only available in Serenity and when the \"with_quality_scores\" query parameter is set to \"true\")",
>                   "x-immutable": true,
>                   "x-read-only": true
1840,1851c1878,1897
<                 "quality_scores": {
<                   "print": {
<                     "en_US": "A",
<                     "fr_FR": "B"
<                   },
<                   "mobile": {
<                     "en_US": "C",
<                     "fr_FR": "D"
<                   },
<                   "ecommerce": {
<                     "en_US": "A",
<                     "fr_FR": "E"
---
>                 "quality_scores": [
>                   {
>                     "scope": "ecommerce",
>                     "locale": "en_US",
>                     "data": "A"
>                   },
>                   {
>                     "scope": "ecommerce",
>                     "locale": "fr_FR",
>                     "data": "B"
>                   },
>                   {
>                     "scope": "tablet",
>                     "locale": "en_US",
>                     "data": "D"
>                   },
>                   {
>                     "scope": "tablet",
>                     "locale": "fr_FR",
>                     "data": "E"
1853c1899
<                 }
---
>                 ]
2015c2061
<             "description": "Return product's quality scores in the response.",
---
>             "description": "Return product quality scores in the response. (Only available in the PIM Serenity version.)",
2066c2112
<                   "description": "Code of the parent product model when the product is a variant (only available since the 2.0). This parent can be modified since the 2.3.",
---
>                   "description": "Code of the parent product model when the product is a variant (only available since the 2.0). This parent can be modified since the 2.3. In Serenity and from the 5.0, the parent of an existing variant product can be set to null: the variant product will become a simple product (with no parent)",
2221c2267,2269
<                   "description": "Only available when query parameter \"with_quality_scores\" is set to \"true\""
---
>                   "description": "Product quality scores for each channel/locale combination (only available in Serenity and when the \"with_quality_scores\" query parameter is set to \"true\")",
>                   "x-immutable": true,
>                   "x-read-only": true
2365,2376c2413,2432
<                 "quality_scores": {
<                   "print": {
<                     "en_US": "A",
<                     "fr_FR": "B"
<                   },
<                   "mobile": {
<                     "en_US": "C",
<                     "fr_FR": "D"
<                   },
<                   "ecommerce": {
<                     "en_US": "A",
<                     "fr_FR": "E"
---
>                 "quality_scores": [
>                   {
>                     "scope": "ecommerce",
>                     "locale": "en_US",
>                     "data": "A"
>                   },
>                   {
>                     "scope": "ecommerce",
>                     "locale": "fr_FR",
>                     "data": "B"
>                   },
>                   {
>                     "scope": "tablet",
>                     "locale": "en_US",
>                     "data": "D"
>                   },
>                   {
>                     "scope": "tablet",
>                     "locale": "fr_FR",
>                     "data": "E"
2378c2434
<                 }
---
>                 ]
2535c2591
<                   "description": "Code of the parent product model when the product is a variant (only available since the 2.0). This parent can be modified since the 2.3.",
---
>                   "description": "Code of the parent product model when the product is a variant (only available since the 2.0). This parent can be modified since the 2.3. In Serenity and from the 5.0, the parent of an existing variant product can be set to null: the variant product will become a simple product (with no parent)",
2690c2746,2748
<                   "description": "Only available when query parameter \"with_quality_scores\" is set to \"true\""
---
>                   "description": "Product quality scores for each channel/locale combination (only available in Serenity and when the \"with_quality_scores\" query parameter is set to \"true\")",
>                   "x-immutable": true,
>                   "x-read-only": true
2834,2845c2892,2911
<                 "quality_scores": {
<                   "print": {
<                     "en_US": "A",
<                     "fr_FR": "B"
<                   },
<                   "mobile": {
<                     "en_US": "C",
<                     "fr_FR": "D"
<                   },
<                   "ecommerce": {
<                     "en_US": "A",
<                     "fr_FR": "E"
---
>                 "quality_scores": [
>                   {
>                     "scope": "ecommerce",
>                     "locale": "en_US",
>                     "data": "A"
>                   },
>                   {
>                     "scope": "ecommerce",
>                     "locale": "fr_FR",
>                     "data": "B"
>                   },
>                   {
>                     "scope": "tablet",
>                     "locale": "en_US",
>                     "data": "D"
>                   },
>                   {
>                     "scope": "tablet",
>                     "locale": "fr_FR",
>                     "data": "E"
2847c2913
<                 }
---
>                 ]
3254c3320
<                   "description": "Code of the parent product model when the product is a variant (only available since the 2.0). This parent can be modified since the 2.3.",
---
>                   "description": "Code of the parent product model when the product is a variant (only available since the 2.0). This parent can be modified since the 2.3. In Serenity and from the 5.0, the parent of an existing variant product can be set to null: the variant product will become a simple product (with no parent)",
3409c3475,3477
<                   "description": "Only available when query parameter \"with_quality_scores\" is set to \"true\""
---
>                   "description": "Product quality scores for each channel/locale combination (only available in Serenity and when the \"with_quality_scores\" query parameter is set to \"true\")",
>                   "x-immutable": true,
>                   "x-read-only": true
3553,3564c3621,3640
<                 "quality_scores": {
<                   "print": {
<                     "en_US": "A",
<                     "fr_FR": "B"
<                   },
<                   "mobile": {
<                     "en_US": "C",
<                     "fr_FR": "D"
<                   },
<                   "ecommerce": {
<                     "en_US": "A",
<                     "fr_FR": "E"
---
>                 "quality_scores": [
>                   {
>                     "scope": "ecommerce",
>                     "locale": "en_US",
>                     "data": "A"
>                   },
>                   {
>                     "scope": "ecommerce",
>                     "locale": "fr_FR",
>                     "data": "B"
>                   },
>                   {
>                     "scope": "tablet",
>                     "locale": "en_US",
>                     "data": "D"
>                   },
>                   {
>                     "scope": "tablet",
>                     "locale": "fr_FR",
>                     "data": "E"
3566c3642
<                 }
---
>                 ]
32612c32688
<           "description": "Code of the parent product model when the product is a variant (only available since the 2.0). This parent can be modified since the 2.3.",
---
>           "description": "Code of the parent product model when the product is a variant (only available since the 2.0). This parent can be modified since the 2.3. In Serenity and from the 5.0, the parent of an existing variant product can be set to null: the variant product will become a simple product (with no parent)",
32767c32843,32845
<           "description": "Only available when query parameter \"with_quality_scores\" is set to \"true\""
---
>           "description": "Product quality scores for each channel/locale combination (only available in Serenity and when the \"with_quality_scores\" query parameter is set to \"true\")",
>           "x-immutable": true,
>           "x-read-only": true
32911,32922c32989,33008
<         "quality_scores": {
<           "print": {
<             "en_US": "A",
<             "fr_FR": "B"
<           },
<           "mobile": {
<             "en_US": "C",
<             "fr_FR": "D"
<           },
<           "ecommerce": {
<             "en_US": "A",
<             "fr_FR": "E"
---
>         "quality_scores": [
>           {
>             "scope": "ecommerce",
>             "locale": "en_US",
>             "data": "A"
>           },
>           {
>             "scope": "ecommerce",
>             "locale": "fr_FR",
>             "data": "B"
>           },
>           {
>             "scope": "tablet",
>             "locale": "en_US",
>             "data": "D"
>           },
>           {
>             "scope": "tablet",
>             "locale": "fr_FR",
>             "data": "E"
32924c33010
<         }
---
>         ]
32990c33076
<               "description": "Code of the parent product model when the product is a variant (only available since the 2.0). This parent can be modified since the 2.3.",
---
>               "description": "Code of the parent product model when the product is a variant (only available since the 2.0). This parent can be modified since the 2.3. In Serenity and from the 5.0, the parent of an existing variant product can be set to null: the variant product will become a simple product (with no parent)",
33145c33231,33233
<               "description": "Only available when query parameter \"with_quality_scores\" is set to \"true\""
---
>               "description": "Product quality scores for each channel/locale combination (only available in Serenity and when the \"with_quality_scores\" query parameter is set to \"true\")",
>               "x-immutable": true,
>               "x-read-only": true
33289,33300c33377,33396
<             "quality_scores": {
<               "print": {
<                 "en_US": "A",
<                 "fr_FR": "B"
<               },
<               "mobile": {
<                 "en_US": "C",
<                 "fr_FR": "D"
<               },
<               "ecommerce": {
<                 "en_US": "A",
<                 "fr_FR": "E"
---
>             "quality_scores": [
>               {
>                 "scope": "ecommerce",
>                 "locale": "en_US",
>                 "data": "A"
>               },
>               {
>                 "scope": "ecommerce",
>                 "locale": "fr_FR",
>                 "data": "B"
>               },
>               {
>                 "scope": "tablet",
>                 "locale": "en_US",
>                 "data": "D"
>               },
>               {
>                 "scope": "tablet",
>                 "locale": "fr_FR",
>                 "data": "E"
33302c33398
<             }
---
>             ]
39273c39369
<       "description": "Return product's quality scores in the response.",
---
>       "description": "Return product quality scores in the response. (Only available in the PIM Serenity version.)",
```

meganelepalud-akeneo commented:
Validated on my side. 
I'm adding some updates to the to-do list related to the 5.0 release. 

It could be a little bit confusing to have this kind of description
`"description": "Code of the parent product model when the product is a variant (only available since the 2.0). This parent can be modified since the 2.3. In Serenity and from the 5.0, the parent of an existing variant product can be set to null: the variant product will become a simple product (with no parent)"`

nmarniesse commented:
> Validated on my side.
> I'm adding some updates to the to-do list related to the 5.0 release.
> 
> It could be a little bit confusing to have this kind of description
> `"description": "Code of the parent product model when the product is a variant (only available since the 2.0). This parent can be modified since the 2.3. In Serenity and from the 5.0, the parent of an existing variant product can be set to null: the variant product will become a simple product (with no parent)"`

Thanks Megane. Strange I removed this description yesterday and added a paragraph in the product section instead
